### PR TITLE
[feature](Azure) Implement azure table value function

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Formerly we forbid using azure as s3 tvf's backend storage, this pr implements the corresponding logic. User should specify the provider property to azure when using azure tvf.

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java [hunk 1]
```diff
@@ -113,6 +109,26 @@ public S3TableValuedFunction(Map<String, String> properties) throws AnalysisExce
         }
     }
 
+    private String constructEndpoint(Map<String, String> properties, S3URI s3uri) throws AnalysisException {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java:112]: Do we have to check ak sk for all kinds of providers?

Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java:114]: No need to do this.

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by morningman — APPROVED** [review]

**Pull request merged**